volume-providers/vpc/provider/delete_volume.go

```diff
@@ -54,3 +58,25 @@ func validateVolume(volume *provider.Volume) (err error) {
 	err = userError.GetUserError("InvalidVolumeID", nil, volume.VolumeID)
 	return
 }
+
+// WaitForVolumeDeletion checks the volume for valid status
+func WaitForVolumeDeletion(vpcs *VPCSession, volumeID string) (err error) {
+	vpcs.Logger.Debug("Entry of WaitForVolumeDeletion method...")
+	defer vpcs.Logger.Debug("Exit from WaitForVolumeDeletion method...")
+
+	vpcs.Logger.Info("Getting volume details from VPC provider...", zap.Reflect("VolumeID", volumeID))
+
+	err = vpcs.APIRetry.FlexyRetry(vpcs.Logger, func() (error, bool) {
+		_, err = vpcs.Apiclient.VolumeService().GetVolume(volumeID, vpcs.Logger)
+		// Keep retry, until GetVolume returns volume not found
+		if err != nil {
+			return err, skipRetryForDeleteVolume(err)
```

Comment:
Two points here.
1- we need to add RIaaS error code `volume_id_not_found` also in case of volume not found which shows that either its deleted or volume not found. 
2- I think we can directly use existing method i.e skipRetry here after typecast or call `skipRetry` inside `skipRetryForDeleteVolume` as most of the code is duplicate.